spring-cloud-kubernetes-fabric8-discovery/src/main/java/org/springframework/cloud/kubernetes/fabric8/discovery/KubernetesDiscoveryClientUtils.java

```diff
@@ -37,4 +54,89 @@ static EndpointSubsetNS subsetsFromEndpoints(Endpoints endpoints, Supplier<Strin
 		return new EndpointSubsetNS(clientNamespace.get(), List.of());
 	}
 
+	static int endpointsPort(EndpointSubset endpointSubset, String serviceId, KubernetesDiscoveryProperties properties,
```

Comment:
the logic here is the same as it was, I just added more logs in debug mode and wrote unit tests for it individually.